File: Markets/Python/options.py
"""
P/L -- profit/loss 
Calls (break_even and P/L)
Puts (break_even and P/L) 

Options are made up of: 
- Intrinsic Value 
- Extrinsic Value 

Add the shebang (first line): 
!/usr/bin/env python3
Make it executable: 
chmod +x options.py 
Run it with: 
./options.py 

Variables: 
S - current price
K - strike price
cprice - call contract price 
pprice - put contract price
call_be - break even call 
put_be - break even put
"""
class Options: 
    def __init__(self, S, K):
        self.S = S
        self.K = K
    def call_be(self, cprice):
        self.call_be = self.K + cprice
        return self.call_be
    def put_be(self, pprice): 
        self.put_be = self.K - pprice
        return self.put_be
    def call(self, end):
        profitability = []
        prices = []
        for i in range(self.S, end+5, 5): 
            prices.append(i) 
            profitability.append(round(max(0, i - self.call_be), 2))
        ans = zip(prices, profitability) 
        print("P/L Call Options")
        for i,j in ans:
            print(i, j)
        #return print(f"P/L Call Option: ", *zip(prices, profitability))
    def put(self, end): 
        profitability = []
        prices = []
        for i in range(self.S, end+5, 5): 
            prices.append(i)
            profitability.append(round(max(0, self.put_be - i)))
        ans = zip(prices, profitability) 
        print("P/L Put Options")
        for i,j in ans: 
            print(i, j)
        #return print(f"P/L Put Option: ", *zip(prices, profitability)) 

# Tests 
current_price = 110 
strike_price = 90

#creating an object -- test
test = Options(current_price, strike_price)

#cprice -- contract price per contract (each contract 100 shares)
cprice = 1.20 

#end value for the loop 
end_value = 150 

#answers
print("Break Even: ", test.call_be(cprice)) 
test.call(end_value) 

File: Markets/Go/options.go
// Run with 
// go run options.go

package main

import (
 "fmt"
 "math"
)

func calls(S, E, cprice float64) float64 {
    profitability := math.Max(0, S - (E + cprice))
    return profitability
}

func puts(S, E, cprice float64) float64 {
	profitability := math.Max(0, (E - cprice) - S)
	return profitability
}

func main() {
	//Call test 
	var Sc float64 = 110.00
   	var Ec float64 = 90.00
    	var cpricec float64 = 1.20

    	cans := calls(Sc, Ec, cpricec)

    	fmt.Printf("Call Profitability: %.2f ", cans)
	fmt.Printf("\n")

	//Put test
	var S float64 = 90.00
	var E float64 = 100.00
	var cprice float64 = 2.00
	ans := puts(S, E, cprice)

	fmt.Println("Put Profitability: ", ans)
}
